src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -425,19 +429,19 @@ subroutine initialize_ice_shelf_dyn(param_file, Time, ISS, CS, G, US, diag, new_
   ! previously allocated for registration for restarts.
 
   if (active_shelf_dynamics) then
-    allocate( CS%u_bdry_val(Isdq:Iedq,Jsdq:Jedq) ) ; CS%u_bdry_val(:,:) = 0.0
```

Comment:
All of these memory declarations are problematic, and would not work for variables on a C-grid with symmetric memory.  See the comment about about CS%u_shelf and CS%v_shelf for guidance on how to fix it.